packages/dev/core/src/Materials/shaderMaterial.ts

```diff
@@ -792,46 +793,10 @@ export class ShaderMaterial extends PushMaterial {
         }
 
         // Clip planes
-        if ((this._options.useClipPlane === null && !!scene.clipPlane) || this._options.useClipPlane) {
-            defines.push("#define CLIPPLANE");
-            if (uniforms.indexOf("vClipPlane") === -1) {
-                uniforms.push("vClipPlane");
-            }
-        }
-
-        if ((this._options.useClipPlane === null && !!scene.clipPlane2) || this._options.useClipPlane) {
-            defines.push("#define CLIPPLANE2");
-            if (uniforms.indexOf("vClipPlane2") === -1) {
-                uniforms.push("vClipPlane2");
-            }
-        }
-
-        if ((this._options.useClipPlane === null && !!scene.clipPlane3) || this._options.useClipPlane) {
-            defines.push("#define CLIPPLANE3");
-            if (uniforms.indexOf("vClipPlane3") === -1) {
-                uniforms.push("vClipPlane3");
-            }
-        }
+        if (this._options.useClipPlane !== false) {
+            addClipPlaneUniforms(uniforms);
 
-        if ((this._options.useClipPlane === null && !!scene.clipPlane4) || this._options.useClipPlane) {
-            defines.push("#define CLIPPLANE4");
-            if (uniforms.indexOf("vClipPlane4") === -1) {
-                uniforms.push("vClipPlane4");
-            }
-        }
-
-        if ((this._options.useClipPlane === null && !!scene.clipPlane5) || this._options.useClipPlane) {
-            defines.push("#define CLIPPLANE5");
-            if (uniforms.indexOf("vClipPlane5") === -1) {
-                uniforms.push("vClipPlane5");
-            }
-        }
-
-        if ((this._options.useClipPlane === null && !!scene.clipPlane6) || this._options.useClipPlane) {
-            defines.push("#define CLIPPLANE6");
-            if (uniforms.indexOf("vClipPlane6") === -1) {
-                uniforms.push("vClipPlane6");
-            }
+            prepareDefinesForClipPlanes(this, scene, defines);

```

Comment:
You should pass `this._options.useClipPlane` to `prepareDefinesForClipPlanes`.

Reply:
This has been factored out two lined above with !== false ?